src/algorithms/acquisition/adapters/beidou_b2ad_pcps_acquisition.cc

```diff
@@ -0,0 +1,324 @@
+    dump_filename_ = configuration_->property(role + ".dump_filename", default_dump_filename);
+    acq_parameters.dump_filename = dump_filename_;
+    //--- Find number of samples per spreading code -------------------------
+    code_length_ = static_cast<unsigned int>(std::round(static_cast<double>(fs_in_) / (BEIDOU_B2ad_CODE_RATE_HZ / static_cast<double>(BEIDOU_B2ad_CODE_LENGTH_CHIPS))));
+
+    vector_length_ = code_length_;
+
+    if (bit_transition_flag_)
+        {
+            vector_length_ *= 2;
+        }
+
+    code_ = new gr_complex[vector_length_];
+
+    if (item_type_.compare("cshort") == 0)
+        {
+            item_size_ = sizeof(lv_16sc_t);
+        }
+    else
+        {
+            item_size_ = sizeof(gr_complex);
+        }
+    acq_parameters.samples_per_code = code_length_;
+    acq_parameters.samples_per_ms = code_length_;
+    acq_parameters.sampled_ms = 1;// TODO sara adjust for NH;
+    acq_parameters.it_size = item_size_;
+    acq_parameters.num_doppler_bins_step2 = configuration_->property(role + ".second_nbins", 4);
+    acq_parameters.doppler_step2 = configuration_->property(role + ".second_doppler_step", 125.0);
+    acq_parameters.make_2_steps = configuration_->property(role + ".make_two_steps", false);
+    acquisition_ = pcps_make_acquisition(acq_parameters);
+    DLOG(INFO) << "BEIDOU acquisition(" << acquisition_->unique_id() << ")";
+
+    stream_to_vector_ = gr::blocks::stream_to_vector::make(item_size_, vector_length_);
+    DLOG(INFO) << "stream_to_vector(" << stream_to_vector_->unique_id() << ")";
+
+    if (item_type_.compare("cbyte") == 0)
+        {
+            cbyte_to_float_x2_ = make_complex_byte_to_float_x2();
+            float_to_complex_ = gr::blocks::float_to_complex::make();
+        }
+
+    channel_ = 0;
+    threshold_ = 0.0;
+    doppler_step_ = 0;
+    gnss_synchro_ = 0;
+}
+
+
+BeidouB2adPcpsAcquisition::~BeidouB2adPcpsAcquisition()
+{
+    delete[] code_;
+}
+
+
+void BeidouB2adPcpsAcquisition::set_channel(unsigned int channel)
+{
+    channel_ = channel;
+    acquisition_->set_channel(channel_);
+}
+
+
+void BeidouB2adPcpsAcquisition::set_threshold(float threshold)
+{
+    float pfa = configuration_->property(role_ + boost::lexical_cast<std::string>(channel_) + ".pfa", 0.0);
+
+    if (pfa == 0.0)
+        {
+            pfa = configuration_->property(role_ + ".pfa", 0.0);
+        }
+    if (pfa == 0.0)
+        {
+            threshold_ = threshold;
+        }
+    else
+        {
+            threshold_ = calculate_threshold(pfa);
+        }
+
+    DLOG(INFO) << "Channel " << channel_ << " Threshold = " << threshold_;
+
+    acquisition_->set_threshold(threshold_);
+}
+
+
+void BeidouB2adPcpsAcquisition::set_doppler_max(unsigned int doppler_max)
+{
+    doppler_max_ = doppler_max;
+
+    acquisition_->set_doppler_max(doppler_max_);
+}
+
+
+// Be aware that Doppler step should be set to 2/(3T) Hz, where T is the coherent integration time (GPS L2 period is 0.02s)
```

Comment:
Please keep comments in doxygen format and add those to .h files as \note or equivalent. Or remove old ones